core/src/main/java/org/elasticsearch/transport/netty/NettyTransportChannel.java

```diff
@@ -81,6 +87,9 @@ public void sendResponse(TransportResponse response) throws IOException {
 
     @Override
     public void sendResponse(TransportResponse response, TransportResponseOptions options) throws IOException {
+        if (reservation != null) {
+            reservation.doReturn();
```

Comment:
My rationale was to ensure that we return the reservation at this point no matter what. For example, a few lines below there is:

```java
ReleasableBytesStreamOutput bStream = new ReleasableBytesStreamOutput(transport.bigArrays);
```
which can throw a `CircuitBreakingException` (outside the `try`). I have restructured the code so the stream is created in the `try`.